### PR TITLE
Game health

### PR DESCRIPTION
**Pull request by javiguerra777**
## Changes
1. Added player health and lives to game
2. The UI displays the players health and lives
3. Added endgame scene that plays when player losses all lives and health
4. Endgame scene lets the player restart the game.
5. Moved any global variables from game to the scene class

## Purpose
The purpose of this feature is to allow an endgame to occur if a player loses. The endgame scene is built to allow to restart and play the game again.

## Approach
This change finally gives an endgame to the game so that there is a competitive aspect to this game.

## Learning
Did some reading on the documentation on phasers website to allow the ability to switch scenes. I also read an article online on how to set it up to allow multiple scenes per game.

https://www.thepolyglotdeveloper.com/2020/09/switch-between-scenes-phaser-game/

Closes issues with global variable within the game.

### CODE REVIEW

File: src/components/SinglePlayerGame.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React, { useEffect } from 'react';
```
Comment (frankieavina) [inline src/components/SinglePlayerGame.tsx:1]: So I saw in your game component you deleted this Single Player mode. Are you still going to incorporate this ?

File: src/pages/Game.tsx [hunk 2]
```diff
@@ -62,6 +64,17 @@ function Game() {
       socket.emit('leave_room', id);
     };
   }, [id]);
+  useEffect(() => {
+    dispatch(setId(id));
```
Comment (frankieavina) [inline src/pages/Game.tsx:68]: little confused as to what is going on here. You set the id then do a callback to set id to empty?

File: src/phaser-game/scenes/EndGame.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import Phaser from 'phaser';
+
+class EndGame extends Phaser.Scene {
```
Comment (frankieavina) [inline src/phaser-game/scenes/EndGame.ts:3]: uuu you're using class to build this interesting

**Review by frankieavina — APPROVED** [review]
Good job overall. Hopefully you can figure it out. Project looks promising

File: src/components/SinglePlayerGame.tsx (hunk 1 above)
Comment (javiguerra777) [reply src/components/SinglePlayerGame.tsx:1]: Yes, just moved it into a different component.

File: src/pages/Game.tsx (hunk 2 above)
Comment (javiguerra777) [reply src/pages/Game.tsx:68]: Yes, we set the ID to the params when the component loads, and when we leave the component the id is empty this helps with changing the ID for the player room

File: src/phaser-game/scenes/EndGame.ts (hunk 3 above)
Comment (javiguerra777) [reply src/phaser-game/scenes/EndGame.ts:3]: Yes, I know OOP is controversial, but the Phaser documentation is written with classes so wish me luck lol

**Pull request merged**